src/main/java/ai/shreds/Adapter/AdapterProductController/AdapterProductController.java

```diff
@@ -0,0 +1,91 @@
+package adapters.primary;
+
+import com.example.adapters.primary.AdapterExceptionHandler;
+import com.example.application.ApplicationProductServicePort;
+import com.example.shared.SharedCreateProductRequest;
+import com.example.shared.SharedUpdateProductRequest;
+import com.example.shared.SharedListProductsParams;
+import com.example.shared.SharedProductDTO;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import java.util.List;
+
+@RestController
+@RequestMapping("/products")
+public class AdapterProductController {
```

Comment:
Remove All exceptions handling from the controller as they are handled by the AdapterExceptionHandler. make sur the Exceptions Class existe. if not propose to create them